File: proxy.php
<?php
header("Access-Control-Allow-Origin: *");
header("Access-Control-Allow-Headers: *");
header("Content-Type: application/json");

$API_KEY = "AIzaSyB12M3tR7J9R9HoHKB9SHwbEJHf2I4pM10";  // Replace with your real API key
$MODEL   = "gemini-2.5-flash";

// Build URL to Google API
$url = "https://generativelanguage.googleapis.com/v1beta/models/$MODEL:generateContent?key=$API_KEY";

// Receive JSON from InfinityFree
$payload = file_get_contents("php://input");

// Prepare HTTP request to forward to Google API
$context = stream_context_create([
    "http" => [
        "method"  => "POST",
        "header"  => "Content-Type: application/json",
        "content" => $payload
    ]
]);

// Send the request and get the response from Google AI
$response = file_get_contents($url, false, $context);

// Return response to InfinityFree
echo $response;

File: proxy.py
from flask import Flask, request, jsonify
import requests

app = Flask(__name__)

API_KEY = "AIzaSyB12M3tR7J9R9HoHKB9SHwbEJHf2I4pM10"  # Replace with your actual API key
MODEL = "gemini-2.5-flash"

@app.route('/proxy', methods=['POST'])
def proxy():
    # Get the incoming request data (the user message)
    user_input = request.get_json()

    # Google API URL
    url = f"https://generativelanguage.googleapis.com/v1beta/models/{MODEL}:generateContent?key={API_KEY}"

    # Make a POST request to Google API with the user input
    response = requests.post(url, json=user_input)

    # If the response is OK, return it to InfinityFree
    if response.status_code == 200:
        return jsonify(response.json())
    else:
        return jsonify({"error": "Error contacting Google API."}), 500

if __name__ == '__main__':
    app.run(debug=True, host="0.0.0.0", port=10000)
